style.md

```diff
@@ -1894,6 +1896,172 @@ this contract. Specifying field names inside tags makes the contract explicit,
 and it guards against accidentally breaking the contract by refactoring or
 renaming fields.
 
+### Don't fire-and-forget goroutines
+
+Goroutines are lightweight, but they're not free. They cost memory.
+Leaking goroutines slowly consume more and more memory.
+
+Therefore, do not leak goroutines in production code.
+Use [go.uber.org/goleak](https://pkg.go.dev/go.uber.org/goleak)
+to test for goroutine leaks inside packages that may spawn goroutines.
+
+In general, every goroutine:
+
+- must have a predictable time at which it will stop running; or
+- there must be a way to signal to the goroutine that it should stop
+
+In both cases, there must be a way code to block and wait for the goroutine to
+finish.
+
+For example:
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+go func() {
+  for {
+    flush()
+    time.Sleep(delay)
+  }
+}()
+```
+
+</td><td>
+
+```go
+stop := make(chan struct{})
+go func() {
+  ticker := time.NewTicker(delay)
+  defer ticker.Stop()
+  for {
+    select {
+    case <-tick.C:
+      flush()
+    case <-stop:
+      return
+    }
+  }
+}()
+```
```

Comment:
what do you think of expanding the example a little to show a `Stop` method as well? I often see implementations where there's a `close(stop)`, but I usually recommend that the stop method waits for the background goroutine to complete.

Reply:
My original intent was that this example only talks about how to build the ability to stop a goroutine into it,
and the next section talks about waiting for the goroutine to exit with either a WaitGroup or a channel. 

Let me see what else I can adjust here to make that clearer.

Reply:
Okay, seeing as I already say "there must be a predictable lifetime" and "in both cases there must be a way to wait for exit", I'll incorporate waiting for the exit in here similar to @mway's suggestion but with a channel instead of a wait group.

in the other sample where i talk about waiting for goroutines, i'll present both options.